### ISSUE TITLE
Flux Optimizers should define equality

### ISSUE THREAD

**Issue by oxinabox** [FluxML/Flux.jl#823]
I was writing some tests, and I wanted to be sure that the hyper-parameters of two models were the same.
So I was checking equality on an object that contained all my hyperparameters,
including the optimizer (which has its own hyper-parameters).

Turns out that fails as
```
julia> Flux.Optimise.NADAM(0.003, (0.9, 0.9), IdDict{Any,Any}()) == Flux.Optimise.NADAM(0.003, (0.9, 0.9), IdDict{Any,Any}())
false
```

**Comment by dhairyagandhi96** [comment]
Should be much easier as we remove the states completely from the optimisers.

**Comment by MikeInnes** [comment]
Yeah, this is not well defined right now but can happen as part of #637.

**Comment by ToucheSir** [comment]
#637 has since been spun off as Optimisers.jl, which supports equality comparison OOTB :)
